cms/djangoapps/contentstore/asset_storage_handlers.py

```diff
@@ -714,7 +715,21 @@ def get_asset_json(display_name, content_type, date, location, thumbnail_locatio
     Helper method for formatting the asset information to send to client.
     '''
     asset_url = StaticContent.serialize_asset_key_with_slash(location)
-    external_url = urljoin(configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL), asset_url)
+    lms_root = configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
+
+    try:
+        ## .. filter_implemented_name: LmsUrlCreationStarted
+        ## .. filter_type: org.openedx.course_authoring.lms.url.creation.started.v1
+        lms_root = LmsUrlCreationStarted.run_filter(
+            context=lms_root,
+            org=location.org,
+            val_name='LMS_ROOT_URL',
+            default=settings.LMS_ROOT_URL
```

Comment:
According to the documentation and definitions we set for the framework, each filter must return the same arguments it received: https://docs.openedx.org/projects/openedx-filters/en/latest/decisions/0003-hooks-filter-tooling-pipeline.html#consequences